### PR TITLE
[KASAN] Extract common MD code.

### PR DESCRIPTION
**Pull request by MichalBlk**
It turns out that architecture-dependent code consists of only two defines: start of the shadow memory and size of the sanitized address. The remaining values are computed using the aforementioned macros.

Besides, this PR cleans up the code. This includes removing the redundant `kasan_va_to_shadow` function that has exactly the same goal as `addr_to_shad` (previously `kasan_md_addr_to_shad`), i.e. compute the address of the shadow byte describing the provided sanitized memory area.

### CODE REVIEW

File: sys/kern/kasan.c [hunk 1]
```diff
@@ -222,17 +236,12 @@ static void call_ctors(void) {
   }
 }
 
-static inline vaddr_t kasan_va_to_shadow(vaddr_t va) {
-  return KASAN_MD_SHADOW_START +
-         (va - KASAN_MD_SANITIZED_START) / KASAN_SHADOW_SCALE_SIZE;
-}
```
Comment (pj1031999) [inline sys/kern/kasan.c:228]: It's not obvious why it chan be replaced by `addr_to_shad` if you look into that first time. Could you add a comment to commit message with explanation?

**Review by pj1031999 — APPROVED** [review]

File: sys/kern/kasan.c [hunk 2]
```diff
@@ -299,7 +308,6 @@ void __asan_unregister_globals(struct __asan_global *globals, size_t n) {
   /* never called */
 }
 
-/* Note: alloca is currently used in strntoul and test_sleepq_sync functions */
```
Comment (jpszczolowski) [inline sys/kern/kasan.c:302]: Is that no longer true?

**Review by jpszczolowski — APPROVED** [review]

File: sys/kern/kasan.c (hunk 2 above)
Comment (MichalBlk) [reply sys/kern/kasan.c:302]: Yes, this message is obsolete.

Comment (jpszczolowski) [reply sys/kern/kasan.c:302]: Just FYI, this wasn’t about us explicitly using alloca in the implementation but about the compiler using alloca while compiling those functions

File: sys/kern/kasan.c (hunk 1 above)
Comment (MichalBlk) [reply sys/kern/kasan.c:228]: Sure. I extended the description of the PR to explain the removal of `kasan_va_to_shadow`. I'll also tell @cahirwpz to mention it in the comment message while doing squash and merge.

**Pull request merged**